### DISCUSSION TITLE
TripsLayer with custom data not displaying

### DISCUSSION THREAD

**Discussion by ventoline** [visgl/deck.gl#8122] [Q&A] [answered]
`
   new TripsLayer({

      id: 'trips',
     data: DrawPositions ,//
      getPath: d => [d.x, d.y] ,// DrawPositions,[-122.69188148947751, 49.32957537190202],    [-122.6923964736084, 49.32979911722749]    [-122.6927397963622, 49.32968724469144], [-122.69359810324714, 49.3296872446922],
      getTimestamps: TimeStamps, // [7.10, 15.50, 24.20, 31.90, 39.90, 47.20, 55.60, 63...
      getColor: [155, 255, 180, 200],
      opacity: 0.9,
      widthMinPixels: 20,
      jointRounded: true,
      trailLength: 280,
      fadeTrail: true,
      currentTime,
      shadowEnabled: false ,
    } ) 
`

Anything I am missing?

**Comment by Pessimistress** [comment]
https://deck.gl/docs/api-reference/layers/path-layer#getpath

**Reply by ventoline** [reply]
thanks, i fixed the vertices format, something else is amiss. status is "discarded: awaiting garbage collection"
`

   new TripsLayer({

      id: 'trips',
     getPath: vertices, // [x0, y0, x1, y1, x2, y2, ...]
     positionFormat: `XY`,
      getTimestamps: TimeStamps,
      getColor: [155, 255, 180, 200],
      opacity: 0.9,
      widthMinPixels: 20,
      jointRounded: true,
      trailLength: 280,
      fadeTrail: true,
      currentTime: 200,
      shadowEnabled: false ,
    } ) `
![image](https://github.com/visgl/deck.gl/assets/3739983/adce544a-4f34-4da5-9d73-250c0bc3d61a)
![image](https://github.com/visgl/deck.gl/assets/3739983/048f24c0-3ca6-452c-944d-c2a19a8299e9)

**Reply by Pessimistress** [reply]
You are not providing a value to `data`. I suggest you start by reading [how to use a layer](https://deck.gl/docs/developer-guide/using-layers).

Do not look at the internal states of a layer; that does not have anything to do with your issue.

**Reply by ventoline** [reply, marked as answer]
Thank you, i had duplicate layers with the same id and 'data' is now properly set. 
the Layer displays (now there are issue with the coordinate projection, but this is another thread. `)`

`
  let testLayer =  new TripsLayer({

      id: 'trips_'.concat(Math.random()*10000000),
     data: [{coors: DrawPositions, timestamps: TimeStamps}],/=
     getPath:  d => d.coors, // [x0, y0, x1, y1, x2, y2, ...]
     positionFormat: `XY`,
     coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
     wrapLongitude : false,
      getTimestamps: d => d.timestamps,
      getColor: [155, 255, 180],
      opacity: 0.9,
      widthMinPixels: 20,
      jointRounded: true,
      trailLength: 280,
      fadeTrail: true, 
       currentTime ,
      shadowEnabled: false ,
    } ) 
`
